ibc-clients/ics07-tendermint/src/client_state/validation.rs

```diff
@@ -38,7 +42,7 @@ where
             ctx,
             client_id,
             client_message,
-            &DefaultVerifier,
+            &ProdVerifier::default(),
```

Comment:
I am a bit confused about how I would reuse this interface for a custom verifier. After I implement `Verifier` trait for my custom logic, how do I pass it here instead of `ProdVerifier::default()`?

Reply:
See if this works. [b8244a5](https://github.com/cosmos/ibc-rs/pull/1168/commits/b8244a5dd50181f392a5ed70a88d0603da99ab8e)

Reply:
Am I correct to conclude that someone has to rewrite the `ClientState` from scratch?

Reply:
> Am I correct to conclude that someone has to create a wrapper struct

That's right

> Can we add a generic to the current ClientState

Our plan was to do this, but adding a generic introduces complications that eventually lead to refactoring some of the APIs as well. Aside from that, we didn't want to introduce complexity to normal users.
We can consider the generic or even writing macros in the future if there be a serious need.